plugins/org.jkiss.dbeaver.net.ssh.ui/src/org/jkiss/dbeaver/ui/net/ssh/SSHTunnelConfiguratorUI.java

```diff
@@ -285,15 +295,41 @@ protected void execute(@NotNull DBRProgressMonitor monitor) throws Throwable {
                     monitor.subTask("Initialize tunnel");
                     String authTypeName = configuration.getStringProperty("authType");
                     SSHConstants.AuthType authType = CommonUtils.valueOf(SSHConstants.AuthType.class, authTypeName);
-                    if (!configuration.isSavePassword()) {
-                        DBPAuthInfo dbpAuthInfo = promptCredentialsDialog(authType, configuration);
+                    if (!configuration.isSavePassword() && tunnel.getRequiredCredentials(configuration, null) != DBWTunnel.AuthCredentials.NONE) {
+                        DBPAuthInfo dbpAuthInfo = promptCredentialsDialog(authType, configuration.getUserName(),
+                            configuration.getPassword());
                         if (dbpAuthInfo != null) {
                             if (authType.equals(SSHConstants.AuthType.PASSWORD)) {
                                 configuration.setUserName(dbpAuthInfo.getUserName());
                             }
                             configuration.setPassword(dbpAuthInfo.getUserPassword());
                         }
                     }
+                    if (configuration.getBooleanProperty(getJumpServerSettingsPrefix() + "enabled")) { //$NON-NLS-1$
```

Comment:
What this code does? We already load secure props, arent's we?

Reply:
Checks if the jump server is enabled, if it is we probably should summon a password dialog for it. Then writes new creds for jump server